WireAPI/Sources/WireAPI/APIs/TeamsAPI/TeamsAPIV0.swift

```diff
@@ -0,0 +1,289 @@
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: TeamResponseV0.self)
+            .failure(code: 404, error: TeamsAPIError.invalidTeamID)
+            .failure(code: 404, label: "no-team", error: TeamsAPIError.teamNotFound)
+            .parse(response)
+    }
+
+    // MARK: - Get team roles
+
+    func getTeamRoles(for teamID: Team.ID) async throws -> [ConversationRole] {
+        let request = HTTPRequest(
+            path: "\(basePath(for: teamID))/conversations/roles",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: ConversationRolesListResponseV0.self)
+            .failure(code: 403, label: "no-team-member", error: TeamsAPIError.selfUserIsNotTeamMember)
+            .failure(code: 404, error: TeamsAPIError.teamNotFound)
+            .parse(response)
+    }
+
+    // MARK: - Get team members
+
+    func getTeamMembers(
+        for teamID: Team.ID,
+        maxResults: UInt
+    ) async throws -> [TeamMember] {
+        var components = URLComponents(string: "\(basePath(for: teamID))/members")
+        components?.queryItems = [URLQueryItem(name: "maxResults", value: "2000")]
+
+        guard let path = components?.url?.absoluteString else {
+            throw TeamsAPIError.failedToGenerateRequest
+        }
+
+        let request = HTTPRequest(
+            path: path,
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: TeamMemberListResponseV0.self)
+            .failure(code: 400, error: TeamsAPIError.invalidQueryParmeter)
+            .failure(code: 403, label: "no-team-member", error: TeamsAPIError.selfUserIsNotTeamMember)
+            .failure(code: 404, error: TeamsAPIError.teamNotFound)
+            .parse(response)
+    }
+
+    // MARK: - Get legalhold status
+
+    func getLegalholdStatus(
+        for teamID: Team.ID,
+        userID: UUID
+    ) async throws -> LegalholdStatus {
+        let request = HTTPRequest(
+            path: "\(basePath(for: teamID))/legalhold/\(userID.transportString())",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: LegalholdStatusResponseV0.self)
+            .failure(code: 404, error: TeamsAPIError.invalidRequest)
+            .failure(code: 404, label: "no-team-member", error: TeamsAPIError.teamMemberNotFound)
+            .parse(response)
+    }
+
+}
+
+struct TeamResponseV0: Decodable, ToAPIModelConvertible {
+
+    let id: UUID
+    let name: String
+    let creator: UUID
+    let icon: String
+    let icon_key: String?
```

Comment:
Should we use snake case for vars here? We have the optionns set the ison decoder [`keydecodingstrategy`](https://developer.apple.com/documentation/foundation/jsondecoder/keydecodingstrategy/convertfromsnakecase) or custom [`CodingKey`](https://developer.apple.com/documentation/swift/codingkey).

Reply:
Can't we configure it for each endpoint/decoding?

Reply:
Do you mean something like for every endpoint?

```swift
let response = try await httpClient.executeRequest(request)
var decoder = JSONDecoder.default
decoder.keyDecodingStrategy = ...

return try ResponseParser(decoder: decoder)
    .success(code: 200, type: LegalholdStatusResponseV0.self)
    .failure(code: 404, error: TeamsAPIError.invalidRequest)
    .failure(code: 404, label: "no-team-member", error: TeamsAPIError.teamMemberNotFound)
    .parse(response)
```

Reply:
done 77f2f3fad0fd7d002b837cc5caddd332ebc3f9a3